src/Faithlife.WebRequests/WebServiceRequestBase.cs

```diff
@@ -329,6 +336,24 @@ private void SetCookie(HttpResponseHeaders headers, Uri requestUri)
 			}
 		}
 
+		private static WebServiceException WrapRequestException(HttpRequestMessage webRequest, Exception exception)
+		{
+			(exception as WebException).Response?.Dispose();
+			return new WebServiceException(message: "Error building request.", requestMethod: webRequest.Method.Method, requestUri: webRequest.RequestUri, innerException: exception);
+		}
+
+		private static bool ShouldWrapException(Exception exception)
+		{
+			// InvalidDataException can be thrown if GZip stream header returned from server is invalid.
+			// ObjectDisposedException can be thrown by Mono.
+			return
+				exception is WebException ||
```

Comment:
I don't think `HttpClient` is supposed to throw `WebException`, but maybe it still does in exceptional circumstances.

Reply:
I could live with not catching it, to start with. I don't think it should be thrown. I've only ever seen it wrapped inside another exception (`HttpRequestException` or `IOException`).

Reply:
Then I'll delete the `WebException.Response.Dispose` code, too.